keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/LaunchConfigurationResolver.kt

```diff
@@ -0,0 +1,45 @@
+package com.netflix.spinnaker.keel.ec2.resolvers
+
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.ClusterSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.ec2.SPINNAKER_EC2_API_V1
+import com.netflix.spinnaker.keel.plugin.Resolver
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class LaunchConfigurationResolver(private val cloudDriverCache: CloudDriverCache) : Resolver<ClusterSpec> {
+  override val apiVersion: ApiVersion = SPINNAKER_EC2_API_V1
+  override val supportedKind: String = "cluster"
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun invoke(resource: Resource<ClusterSpec>): Resource<ClusterSpec> =
+    resource.run {
+      copy(
+        spec = spec.run {
+          copy(
+            _defaults = defaults.copy(launchConfiguration = defaults.launchConfiguration?.withResolvedKeyPair(
+              spec.locations.account))
+          )
+        }
+      )
+    }
+
+  private fun ClusterSpec.LaunchConfigurationSpec.withResolvedKeyPair(account: String): ClusterSpec.LaunchConfigurationSpec {
+    return if (keyPair == null) {
+      val defaultKeyPair = cloudDriverCache.defaultKeyPairForAccount(account)
+      if (!defaultKeyPair.contains("{{region}}")) {
+        copy(keyPair = defaultKeyPair)
+      } else {
+        // TODO: if it has a placeholder, then instead of setting the default we should set the key pair in the overrides
+        //  for each region with the placeholder replaced, so long as the user has not already specified something.
```

Comment:
I think our external cluster spec should support `{{region}}` expansion so that keypair overrides aren't always necessary when this regional keypair scheme is followed in an account.

Reply:
I thought about that as well, but assumed that setting the corresponding regional overrides in the resolved spec was the (one?) way to achieve that. Should I do that in the `ClusterSpec` `resolve*` methods instead?

Reply:
I think what's here now looks good!